cpp/src/gandiva/like_holder.cc

```diff
@@ -25,7 +25,7 @@ namespace gandiva {
 
 RE2 LikeHolder::starts_with_regex_(R"((\w|\s)*\.\*)");
```

Comment:
You can include characters like - to this list. Same with ends_with optimisation